components/table.tsx

```diff
@@ -1,41 +1,38 @@
 import { HTMLTable } from "@blueprintjs/core";
-import * as React from "react";
-
 import * as styles from "df/components/table.css";
+import * as React from "react";
 
-export type TRow = Array<{
-  cells: Array<React.ReactElement | string>;
+export interface IRow {
+  cells?: Array<React.ReactElement | string>;
   colspans?: number[];
   href?: string;
   onClick?: () => void;
-}>;
+  onMouseEnter?: (event: React.MouseEvent<HTMLAnchorElement, MouseEvent>) => void;
+  onMouseLeave?: (event: React.MouseEvent<HTMLAnchorElement, MouseEvent>) => void;
+}
 
 export interface ITableProps {
   headers: Array<React.ReactElement | string>;
-  rows: TRow;
+  rows: IRow[];
   className?: string;
   condensed?: boolean;
   columnWidths?: number[];
 }
 
-export const Row = ({
-  children,
-  href,
-  onClick
-}: {
-  children: React.ReactElement;
-  href?: string;
-  onClick?: () => void;
-}) => {
-  if (href) {
+export const Row = (
+  row: {
+    children: React.ReactElement;
+  } & IRow
+) => {
+  if (row.href || row.onClick) {
     return (
-      <a className={styles.tableRowLink} onClick={onClick} href={href}>
-        {children}
+      <a className={styles.tableRowLink} {...row}>
```

Comment:
`...row` will add `colspans` to this tag, which is technically invalid. Easy way to remove is to extract colspans from row and then do `...rest`

Reply:
Nice